### PR TITLE
Documentation updates using MyST

### LINKED ISSUE #52: Recommonmark is deprecated: We should switch to MyST
## Context
We are currently using `sphinx` and `ReadtheDocs` to keep our documentation clean and organized. Natively, `sphinx` uses `reStructuredText` (or .rst extension), and the module `recommonmark` has been used to parse `markdown` format. I ​recently found that the module `recommonmark` is no longer maintained, and the developer (and community) shifted to `MyST` parser.

##  Suggestion
I don't think the issue is imminent and we can probably handle the problem in the back, but I can definitely see the community and the supports for the `markdown` + `sphinx` are shifting to `MyST`. It appears to me that `MyST` is now officially supported `markdown` extension for `sphinx`, and we should change our environment as well.

### Change
It looks like `MyST` is fairly easy to use just like `recommonmark`. Most of the change will be done on `conf.py`, and everything should be compatible.

## Notes

[Deprecation note from recommonmark](https://github.com/readthedocs/recommonmark/issues/221)

### PR DESCRIPTION
**Pull request by skim0119**
## Changes

- Remove `recommonmark` and add `MyST` (resolve #52)
  - Add `numpydoc` for parsing numpy-documentation style.
- Rework docs/readme.md file for the developer, including explanation on how `documentations` are made.
- organize documentation files (API, overview)
- minor changes in the class/method description to match `MyST` and `numpydoc`.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -155,7 +155,11 @@ In order to format the code:
 
 ### Documentation styleguide
 
-We follow [NumPy documentation guidelines](https://numpydoc.readthedocs.io/en/latest/format.html) for PyElastica documentation.
+We follow [NumPy documentation guidelines][numpydoc-guideline] for PyElastica documentation.
+If you are interested in contributing or modifying the documentation, please refer [docs-readme][docs-readme] file.
```
Comment (nmnaughton) [inline CONTRIBUTING.md:159]: Typo: should be "please refer *to the* docs-readme file"

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Elastica Documentation
+
+We use [`Sphinx`](https://www.sphinx-doc.org/en/master/) and [`ReadtheDocs`](https://readthedocs.org/) to keep organize our [documentation page](https://docs.cosseratrods.org/en/latest/).
+
+In addition, we utilize following extensions to enhance the documentation :coffee:
```
Comment (nmnaughton) [inline docs/README.md:5]: "we utilize *the* following ..."

File: CONTRIBUTING.md (hunk 1 above)
Comment (skim0119) [reply CONTRIBUTING.md]: Fixed: [78506ca](https://github.com/GazzolaLab/PyElastica/pull/53/commits/78506ca7da513572b56fcc6040e43711981318b8)

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Elastica Documentation
+
+We use [`Sphinx`](https://www.sphinx-doc.org/en/master/) and [`ReadtheDocs`](https://readthedocs.org/) to keep organize our [documentation page](https://docs.cosseratrods.org/en/latest/).
```
[the remark this answers is not in the document]
Comment (skim0119) [reply docs/README.md]: Fixed: 0527d80

File: docs/README.md (hunk 2 above)
Comment (skim0119) [reply docs/README.md]: Fixed: 0527d80

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+make html
+```
+
+Once the documentation building is done, open `docs/_build/html/index.html` to view.
+
+Use `make help` for other options.
+
+# Contribution
+
+The documentation-related commits will be collected in the branch `doc_**` separate from `master` branch, and merged into `master` collectively. Ideally, the updates in the documentation branch will seek up-coming version update (i.e. `update-**` branch), and merged shortly after the version release. If an update is critical and urgent, create PR directly to `master` branch.
```
[the remark this answers is not in the document]
Comment (skim0119) [reply docs/README.md]: Fixed: 0527d80

**Review by nmnaughton — APPROVED** [review]

**Pull request merged**